internal/node/machine_mgr.go

```diff
@@ -112,31 +113,40 @@ func (m *MachineManager) Start() {
 
 	defer func() {
 		if r := recover(); r != nil {
-			m.log.Info(fmt.Sprintf("recovered: %s", r))
+			m.log.Debug(fmt.Sprintf("recovered: %s", r))
 		}
 	}()
 
+	if !m.config.PreserveNetwork {
+		err := m.resetCNI()
+		if err != nil {
+			m.log.Warn("Failed to reset network.", slog.Any("err", err))
+		}
+	}
+
 	for !m.stopping() {
 		select {
 		case <-m.ctx.Done():
 			return
 		default:
+			if len(m.warmVMs) == m.config.MachinePoolSize {
+				time.Sleep(runloopSleepInterval)
+				continue
+			}
+
 			vm, err := createAndStartVM(m.ctx, m.config, m.log)
 			if err != nil {
 				m.log.Warn("Failed to create VMM for warming pool.", slog.Any("err", err))
 				continue
 			}
 
 			go m.awaitHandshake(vm.vmmID)
```

Comment:
if this go routine is non-blocking, the next line adds the VM to the map without verifying a handshake. Is that the behavior we want?

Reply:
Yes because we need to be able to find it later to shut it down, even if it failed the handshake.